Library/Homebrew/utils.rb

```diff
@@ -276,6 +276,39 @@ def self.install_gem_setup_path!(gem, version = nil, executable = gem)
       EOS
     end
   end
+
+  # Hash of Module => Set(method_names)
+  @@injected_dump_stat_modules = {}
+
+  def inject_dump_stats!(the_module, pattern)
+    @@injected_dump_stat_modules[the_module] ||= []
+    injected_methods = @@injected_dump_stat_modules[the_module]
+    the_module.module_eval do
+      instance_methods.grep(pattern).each do |name|
+        next if injected_methods.include? name
+        method = instance_method(name)
+        define_method(name) do |*args, &block|
+          begin
+            time = Time.now
+            method.bind(self).call(*args, &block)
+          ensure
+            $times[name] ||= 0
+            $times[name] += Time.now - time
+          end
+        end
+      end
+    end
+
+    if $times.nil?
+      $times = {}
+      at_exit {
+        col_width = [$times.keys.map(&:size).max + 2, 15].max
+        puts $times.sort_by { |_k, v| v }.map {
```

Comment:
I pulled the block arguments up to the first line, and did the `do ... end` for the `at_exit` block.

But when I switch this particular `{ ... }` to a `do ... end`, the behavior changes. Instead of printing a list of lines

```
Error: 4 problems in 3 formulae
audit_github_repository:    0.0000 sec
audit_style:                0.0000 sec
audit_bottle_spec:          0.0000 sec
audit_conflicts:            0.0000 sec
```

it does a stringification of an Enumerator like this.

```
Error: 4 problems in 3 formulae
#<Enumerator:0x007fb78c6c0d68>
```

Know what's going on there? `ruby` 2.0.0p643 on OS X 10.9.5.